Content.Server/Replays/ReplayEventType.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Content.Server.Replays;
+
+/// <summary>
+/// Used to identify the type of a <see cref="ReplayEvent"/> for external tools to parse the replay
+/// </summary>
+/// <remarks>
+/// Do not rename any of these values, as they are used to identify the type of event in the replay file and renaming them will break compatibility with external tools
+/// Adding new values is fine.
+/// </remarks>
+public enum ReplayEventType : byte
```

Comment:
Why do you need this when the YAML serializer already emits `!type` codes?

Reply:
I use generic classes such as `GenericPlayerEvent` and `GenericObjectEvent` for stuff that doesn't need much more data. In those cases the enum is needed.

Reply:
After thinking about it myself a little bit I do agree that's probably the best way to do it, so this is fine.